hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -186,6 +186,34 @@ object LowerTableIR {
           val env: Env[IR] = Env("row" -> row, "global" -> loweredChild.globals)
           loweredChild.copy(body = StreamMap(loweredChild.body, row.name, Subst(newRow, BindingEnv(env, scan = Some(env)))))
 
+        case t@TableKeyBy(child, newKey, isSorted: Boolean) =>
+          val loweredChild = lower(child)
+          val nPreservedFields = loweredChild.partitioner.kType.fieldNames
+            .zip(newKey)
+            .takeWhile { case (l, r) => l == r }
+            .length
+
+          require(!isSorted || nPreservedFields > 0 || newKey.isEmpty)
```

Comment:
I don't know why this is necessary.  If n = 0, you should just have empty key intervals.

Reply:
this requirement was lifted from RVD. I think it's correct -- if n == 0 (and isSorted == true) then you'll end up with a keyed dataset where you don't have any partitioning information. You can't do anything with that.

Reply:
I can't find any evidence that overlap isn't allowed in RVD.  

I also don't buy n = 0 is special.  If you're coarsening the partitioner, distinct partition endpoints can become the same.  For example: [a 0, a 5], [a 6, a 11] with coarsen(1).  The only thing I could find was this:

 > lteqWithOverlap(allowedOverlap)(left.right, right.left)

which appears to allow equality up to allowed overlap (which appears to be the key length - 1 in the initial case).